massa-protocol-worker/src/node_info.rs

```diff
@@ -133,9 +135,9 @@ impl NodeInfo {
     pub fn insert_known_ops(&mut self, ops: Set<OperationId>, max_ops_nb: usize) {
         for operation_id in ops.into_iter() {
             if self.known_operations.insert(operation_id) {
-                self.known_operations_queue.push_front(operation_id);
-                if self.known_operations_queue.len() > max_ops_nb {
-                    if let Some(op_id) = self.known_operations_queue.pop_back() {
+                self.known_operations_queue.push_back(operation_id);
+                while self.known_operations_queue.len() > max_ops_nb {
+                    if let Some(op_id) = self.known_operations_queue.pop_front() {
```

Comment:
So here you are removing the element that just was inserted. Why the change in logic(versus the `pop_back` used earlier)?

Reply:
The behavior is the same (push_back then pop_front instead of push_front then pop_back).
It is just more canonical with the usual way of doing things with Vec (most recent element / top of the stack to the back).

Reply:
It appears to me the behavior is different, in terms of removing the oldest or newest element when capacity has been reached. I think you could write a test for the initial behavior, which would have failed after the change.